src/frame-writer.cpp

```diff
@@ -171,13 +171,14 @@ void FrameWriter::init_video_stream()
         videoCodecCtx->flags |= AV_CODEC_FLAG_GLOBAL_HEADER;
 
     int err;
+    char str[256];
     if ((err = avcodec_open2(videoCodecCtx, codec, &options)) < 0)
     {
-        std::cerr << "avcodec_open2 failed: " << averr(err) << std::endl;
+        av_strerror(err, str, 256);
```

Comment:
seems like our av_err does exactly this?

Reply:
No, it does not work, averr gives garbage. For example, if I don't set LIBVA_DRIVER_NAME, I hit https://github.com/ammen99/wf-recorder/blob/master/src/frame-writer.cpp#L36 and the terminal shows "Failed to create hw encoding device : Input/ou�w}��y". If I use av_strrerror(), I get this: "Failed to create hw encoding device : Input/output error"

Reply:
Maybe it has something to do with [this comment regarding av_err2str()](https://www.ffmpeg.org/doxygen/4.1/group__lavu__error.html#gac4f6e109c242c81aeee21868d3f35e12): "Convenience macro, the return value should be used only directly in function arguments but never stand-alone." Since you're wrapping it in a function and then using that function, might this be a problem?